### PR TITLE
Add unit/e2e tests for mongo-typeorm sample

### PR DESCRIPTION
**Pull request by kiwikern**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe: Docs / Samples
```

## What is the current behavior?
Issue Number: #1539

## What is the new behavior?
Added unit + e2e tests for mongo+typeorm sample.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: sample/13-mongo-typeorm/test/photos.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import * as request from 'supertest';
+import { AppModule } from '../src/app.module';
+
+describe('Photos (e2e)', () => {
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    const moduleFixture = await Test.createTestingModule({
```
Comment (BrunnerLivio) [inline sample/13-mongo-typeorm/test/photos.e2e-spec.ts]: To be consistent
```suggestion
    const module = await Test.createTestingModule({
```

**Comment by kamilmysliwiec** [comment]
> Update the test:e2e:samples task to make it work with docker-compose out of the box?

How would that look like?

**Comment by BrunnerLivio** [comment]
@kamilmysliwiec Check whether the sample has a `docker-compose.yml` file in the root directory and run `docker-compose up -d` if so.

**Comment by kiwikern** [comment]
Thanks for your feedback @BrunnerLivio. 🙂 I addressed your comments and rebased to master. As expected, the e2e test is now failing since the database is not started.

File: sample/13-mongo-typeorm/src/photo/photo.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Photo } from './photo.entity';
+
+describe('Photo Service', () => {
+  let service: PhotoService;
+  let repository;
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      controllers: [PhotoService],
```
Comment (andrewwongyedpay) [inline sample/13-mongo-typeorm/src/photo/photo.service.spec.ts:12]: PhotoService in controllers?

Comment (kiwikern) [reply sample/13-mongo-typeorm/src/photo/photo.service.spec.ts]: Thanks for your feedback. 🙏 I moved the service into the providers array.

**Comment by kamilmysliwiec** [comment]
Since the only thing the controller does is delegating work to the injected service instance (which we mock in this test), it doesn't really bring any value (does not protect us from anything).

**Comment by kiwikern** [comment]
@kamilmysliwiec I absolutely agree that these tests do not ensure the example code does not break. However, I believe the goal of the tests is to rather serve as examples on how to set up unit/e2e tests for the mongo-sample given. Especially, how you can mock a repository can be of interest for new users.

From the linked issue #1539:
> Give new users like me simple and practical examples for implementing tests

**Pull request closed without merging**